### PR TITLE
fix: Add upper python version restriction to isort and darglint

### LINKED ISSUE #1148: Poetry SolverProblemError
First of all, thanks for writing this amazing template!

I tried to use the template for the creation of a new Python 3.8.5 project. 

Steps I have taken to create the project:
1. `cookiecutter cookiecutter-hypoermodern-python`
2. `poetry lock`

The resulting output I received is:
```
[SolverProblemError]
The current project's Python requirement (>=3.7) is not compatible with some of the required packages Python requirement:
  - isort requires Python >=3.6.1,<4.0

Because isort (5.10.1) requires Python >=3.6.1,<4.0
 and no versions of isort match >5.10.1, isort is forbidden.
So, because some-sample depends on isort (>=5.10.1), version solving failed.
```

Things I tried so far:
- Removing the isort package from the toml --> shifts the problem to other packages
- Removing the lock file before creating the template

### PR DESCRIPTION
**Pull request by eyllanesc**
Fixes #1148

### CODE REVIEW

File: {{cookiecutter.project_name}}/pyproject.toml [hunk 1]
```diff
@@ -28,14 +28,15 @@ click = ">=8.0.1"
 Pygments = ">=2.10.0"
 black = ">=21.10b0"
 coverage = {extras = ["toml"], version = ">=6.2"}
-darglint = ">=1.8.1"
+darglint = {version = ">=1.8.1", python = "<4"}
+urllib3 = {version = ">=1.26.8", python = "<4"}
```
Comment (paw-lu) [inline {{cookiecutter.project_name}}/pyproject.toml:32]: should urllib be here?

Comment (eyllanesc) [reply {{cookiecutter.project_name}}/pyproject.toml:32]: @paw-lu When I tested I saw that if you don't set the version of urllib3 then a version of urllib3 was downloaded that had a bug, and the library that urllib3 used was darglint so I put it there.

Comment (cjolowicz) [reply {{cookiecutter.project_name}}/pyproject.toml:32]: urllib3 should not be here. It requires Python < 4, and it's an indirect dependency via safety and sphinx. W could make this work by adding the environment marker to safety and sphinx as well. However, I'm now considering whether we should add the Python cap back. I'll open an issue shortly so we can discuss the pros and cons of this move.

Comment (cjolowicz) [reply {{cookiecutter.project_name}}/pyproject.toml:32]: See #1154

**Comment by cjolowicz** [comment]
Superceded by https://github.com/cjolowicz/cookiecutter-hypermodern-python-instance/pull/832

**Pull request closed without merging**